### PR TITLE
Addition of test suite for EC2+2@4

### PR DESCRIPTION
**Pull request by pdhiran**
1. Added new module to test EC pool for 2+2 config on a 4 node cluster.
2. Modified other test suites for BM pipeline

Pass logs for EC 2+2 suite on 4 node BM cluster: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-QRQF69

### CODE REVIEW

**Comment by pdhiran** [comment]
Suites not covered in the PR :
1. suites/reef/rados/tier-4_rados_tests.yaml
2. suites/reef/rados/tier-4_rados_test-pool-osd-recovery.yaml
3. suites/reef/rados/tier-2_rados_test_bluestore.yaml
4. suites/reef/rados/tier-2_rados_test-stretch-mode.yaml
5. suites/reef/rados/tier-2_rados_test-slow-op-requests.yaml
6. suites/reef/rados/tier-2_rados_ec-pool_recovery.yaml
7. suites/reef/rados/tier-2_rados_test-pool-functionalities.yaml
8. suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml

**Comment by neha-gangadhar** [comment]
lgtm

File: suites/reef/rados/tier-3_rados_test-4-node-ecpools.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          rados_read_duration: 30
+          test_overwrites_pool: true
+          metadata_pool: re_pool_overwrite
+          image_name: image_ec_pool
+          image_size: 100M
+        delete_pools:
+          - ec_pool_overwrite
+          - re_pool_overwrite
+      desc: EC pool with Overwrites & create RBD pool
+
```
Comment (harshkumarRH) [inline suites/reef/rados/tier-3_rados_test-4-node-ecpools.yaml:142]: Please remove extra blank lines

File: tests/rados/test_four_node_ecpool.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    endtime = datetime.datetime.now() + datetime.timedelta(seconds=10000)
+    while datetime.datetime.now() < endtime:
+        pool_pg_num = rados_obj.get_pool_property(pool=pool_name, props="pg_num")[
+            "pg_num"
+        ]
+        if pool_pg_num == pg_count_bulk_true:
+            log.debug(f"PG count on pool {pool_name} is achieved")
+            break
+        log.debug(
```
Comment (harshkumarRH) [inline tests/rados/test_four_node_ecpool.py:110]: Please consider changing this to log.info for new users to understand where the workflow is waiting to complete

File: tests/rados/test_four_node_ecpool.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    # Stopping all OSDs of 1 host and check for inactive PGs
+    stop_host = osd_nodes[0]
+    osd_list = rados_obj.collect_osd_daemon_ids(osd_node=stop_host)
+    for osd in osd_list:
+        if not rados_obj.change_osd_state(action="stop", target=osd):
+            log.error(f"Unable to stop the OSD : {osd_list[0]}")
+            raise Exception("Execution error")
+        time.sleep(5)
+
+        pool_pgids = rados_obj.get_pgid(pool_name=pool_name)
```
Comment (harshkumarRH) [inline tests/rados/test_four_node_ecpool.py:194]: Please consider moving the check for inactive PGs to a common module. Can be taken up later.

File: tests/rados/test_four_node_ecpool.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+            raise Exception("Execution error")
+        time.sleep(5)
+
+    # tbd: Serviceability scenarios remaining.
+
+    if ec_config.get("delete_pool"):
+        # Deleting the pool created earlier
+        if not rados_obj.detete_pool(pool=pool_name):
+            log.error(f"the pool {pool_name} could not be deleted")
+            return 1
```
Comment (harshkumarRH) [inline tests/rados/test_four_node_ecpool.py:215]: Please consider removing return 1 and adding raise exception if needed

**Review by harshkumarRH — COMMENTED** [review]
LGTM with minor changes

File: suites/reef/rados/tier-3_rados_test-4-node-ecpools.yaml (hunk 1 above)
Comment (pdhiran) [reply suites/reef/rados/tier-3_rados_test-4-node-ecpools.yaml:142]: Done

File: tests/rados/test_four_node_ecpool.py (hunk 3 above)
Comment (pdhiran) [reply tests/rados/test_four_node_ecpool.py:194]: Done.

File: tests/rados/test_four_node_ecpool.py (hunk 2 above)
Comment (pdhiran) [reply tests/rados/test_four_node_ecpool.py:110]: Done

**Comment by pdhiran** [comment]
Hey @neha-gangadhar , @harshkumarRH , @SrinivasaBharath , please review. Have made the requested changes.

**Review by harshkumarRH — APPROVED** [review]
Looks good

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**